### PR TITLE
Spoon to twig converter

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Non critical bugfix

## Pull request description

Fixes the conversion of a theme with the spoon to twig convertor

Also adds some fixes to the actual conversion

- fix conversion spoon_date
- fix conversion substring
- fix geturlforblock
- fix getnavigation
- fix getsubnavigation
- fix words like contact being detected as an action
- added the Mails subdirectory to the directories that need to be scanned
- ucfirst is no longer replaced with capitalize since it is not the same and has been added to fork
- fix converting positions
- fix multiple translations on the same line messing things up

### CODE REVIEW

File: tools/spoon2twig.php [hunk 1]
```diff
@@ -582,10 +582,13 @@ public function ruleParser($filedata)
         $filedata = $this->pregReplaceSprintf('/{{ err(.*?) }}/i', '{{ err.%s }}', $filedata);
         $filedata = $this->pregReplaceSprintf('/{{ act(.*?) }}/i', '{{ act.%s }}', $filedata);
 
-        $filedata = $this->pregReplaceSprintf('/{{ lbl.(.*?)[!^|]/i', "{{ 'lbl.%s'|trans|", $filedata);
-        $filedata = $this->pregReplaceSprintf('/{{ act.(.*?)[!^|]/i', "{{ 'act.%s'|trans|", $filedata);
-        $filedata = $this->pregReplaceSprintf('/{{ msg.(.*?)[!^|]/i', "{{ 'msg.%s'|trans|", $filedata);
-        $filedata = $this->pregReplaceSprintf('/{{ err.(.*?)[!^|]/i', "{{ 'err.%s'|trans|", $filedata);
+        $filedata = $this->pregReplaceSprintf('/{{ lbl.(.*?)[!^|\s}]/i', "{{ 'lbl.%s'|trans|", $filedata);
```
Comment (jessedobbelaere) [inline tools/spoon2twig.php]: 🎉

File: tools/spoon2twig.php [hunk 2]
```diff
@@ -89,6 +89,18 @@ public function start(array $argv)
             }
             $this->convertAllFiles($force, $path, $input);
 
+            $themeInfoXmlPath = $source.'Frontend/Themes/'.$input.'/info.xml';
```
Comment (jeroendesloovere) [inline tools/spoon2twig.php]: @carakas Can you add spaces between variables please?

File: tools/spoon2twig.php [hunk 3]
```diff
@@ -167,6 +236,11 @@ public function convertAllFiles($force, $path, $input = null)
      */
     private function findFiles($BPath, array $path, $input = null)
```
Comment (jeroendesloovere) [inline tools/spoon2twig.php]: $BPath? You mean $basePath?

File: tools/spoon2twig.php (hunk 2 above)
Comment (carakas) [inline tools/spoon2twig.php]: fixed in a5c41ae

File: tools/spoon2twig.php (hunk 3 above)
Comment (carakas) [inline tools/spoon2twig.php]: fixed in 32f0504

**Pull request merged**